### PR TITLE
feat: usePolicyUpdate -> usePolicy for create or edit

### PR DESCRIPTION
**Pull request by akofink**
Policy creation, editing, and viewing should work as expected, including: Policy fields (name, threshold, etc.) business objective, system assignment, and rule assignment.

Resolved issues:

- [x] associateRules is not called yet
- [x] setting business objective is broken
- [x] policy details rules table shows profiles even when `osMinorVersion === ''`

And some that still exist that we should fix, perhaps in a new PR so break up the work:

- error handling for all API calls; we might handle errors in another PR. It's not so trivial since usePolicyUpdate was not written to consider error handling for all API calls (only policy creation mutation)
- not handled: when the profile doesn't exist in the requested benchmark (i.e. 0.1.45 and CIS). This scenario results in a 'missing' tab for those versions where the profile isn't available. We'll need to handle this somehow, probably with COMP-E-178 (filter hosts by policy type).

I've tried to split this work into commits for easy review, but it's going into the feature branch anyway, so we can improve this quite a bit after merge. I mainly want to make sure everything is working as expected.

~Requires https://github.com/RedHatInsights/compliance-backend/pull/804 and https://github.com/RedHatInsights/compliance-backend/pull/803~ (merged)

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/CreatePolicy/FinishedCreatePolicy.js [hunk 1]
```diff
@@ -10,132 +10,80 @@ import { reduxForm, formValueSelector } from 'redux-form';
...
-            }), this.associateSystems);
-        }).catch((error) => {
-            this.setState({
-                message: error.networkError.message,
-                errors: error.networkError.result.errors,
-                failed: true
-            });
-        });
-    }
+const FinishedCreatePolicy = ({
```
Comment (akofink) [inline src/SmartComponents/CreatePolicy/FinishedCreatePolicy.js:15]: class component -> functional component

File: src/SmartComponents/EditPolicy/usePolicyUpdate.js [hunk 2]
```diff
@@ -24,25 +24,44 @@ const useCreateBusinessObjective = () => {
 const usePolicyUpdate = () => {
     const createBusinessObjective = useCreateBusinessObjective();
     const [updateProfile] = useMutation(UPDATE_PROFILE);
+    const [createProfile] = useMutation(CREATE_PROFILE);
     const [associateSystems] = useMutation(ASSOCIATE_SYSTEMS_TO_PROFILES);
 
     return async (policy, updatedPolicy) => {
```
Comment (akofink) [inline src/SmartComponents/EditPolicy/usePolicyUpdate.js:30]: I plan to refactor this function. It's pretty complex right now.

File: src/SmartComponents/EditPolicy/usePolicyUpdate.js [hunk 3]
```diff
@@ -24,25 +24,44 @@ const useCreateBusinessObjective = () => {
...
-            businessObjectiveId: await createBusinessObjective(policy, updatedPolicy?.businessObjective)
+            complianceThreshold: parseFloat(updatedPolicy.complianceThreshold)
         };
 
-        associateSystems({
+        if (businessObjectiveId) {
+            policyInput.businessObjectiveId = businessObjectiveId;
+        }
+
+        if (policy === null) {
```
Comment (vkrizan) [inline src/SmartComponents/EditPolicy/usePolicyUpdate.js]: Nit: `!policy` would do.

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js [hunk 4]
```diff
@@ -17,8 +17,15 @@ SSGTabTitle.propTypes = {
     profile: propTypes.object.isRequired
 };
 
-const PolicyMultiversionRules = ({ policy: { policy: { profiles } } }) => {
-    let tabsData = profiles.map((profile) => ({ profile }));
+const PolicyMultiversionRules = ({ policy: { hosts, policy: { profiles } } }) => {
+    let tabsData = profiles.filter(
+        (profile) => profile.osMinorVersion !== ''
+    ).map((profile) => ({
```
Comment (vkrizan) [inline src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:23]: I'll probably override these changes :-)

**Review by vkrizan — APPROVED** [review]
Looks awesome! Thank you @akofink.

**Pull request merged**